io/genbank/parser_test.go

```diff
@@ -0,0 +1,173 @@
+package genbank
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestParseHeader(t *testing.T) {
+	testcases := []struct {
+		name    string
+		data    string
+		want    Header
+		wantErr bool
+	}{{
+		name: "parses example header from spec",
+		data: `GBBCT1.SEQ          Genetic Sequence Data Bank
```

Comment:
For the cases where `wantErr` is nil, perhaps it could be helpful to define `data` in terms of want. 

So for instance, `Header` could have a method like

```go
func (h Header) bytes() []bytes {
  strbuilder := strings.Builder{}
  for _, r := range h.FileName {
     strbuilder.WriteRune(r)
  }
  for i := 0; i < paddingBeforeTitle-len(h.FileName); i++ {
      strbuilder.WriteByte(" "[0]) // maybe hard code some common byte values for these :shrug: 
   }
  for _, r := range h.Title {
     strbuilder.WriteRune(r)
  }
  strbuilder.WriteByte("\n"[0])
  // etc.
}


	for _, tc := range testcases {
		t.Run(tc.name, func(t *testing.T) {
                        var p *Parser
                        if data == "" {
                            p = NewParser(strings.NewReader(tc.want)
                       } else {
			    p = NewParser(strings.NewReader(tc.want))
                       }
			got, err := p.parseHeader()

			if gotErr := err != nil; gotErr != tc.wantErr {
				t.Fatalf("incorrect error returned from parseHeader, wantErr: %v, err: %v", tc.wantErr, err)
			}

			if diff := cmp.Diff(tc.want, got); !tc.wantErr && diff != "" {
				t.Fatalf("parseHeader returned incorrect header, (-want, +got): %v", diff)
			}

		})
	}
```

This should do 2 very nice things:
1. Contributors don't need to to spend time aligning multi line string literals in accordance to the spec. All they have to do is define a Struct. More complex structs of the Genbank could have some constructors with some nice defaults to help build up the test data if it is too complex to define the struct in place or if we find that it takes too much boiler plate to define over and over again. This should make it easier to add and reason about test data.
2. This gets us part of the way to implementing writing to disk! 

This is probably overkill for header and locus, but maybe good to keep in mind for the future. Having golden tests would be super helpful once we're ready to test some complete Genbank files. This would make it trivial to add test cases and work through any problems if bugs reported.

Reply:
Good points. We need write ability - makes sense to implement this along the way so we can just do round-trip tests! Will go back and do this.

Reply:
Latest commit implements these suggestions for the header:

- separate "happy" and "sad" tests into `TestXXXRoundtrip` and `TestXXXErr`
- make tests roundtrip instead

Still need to do this for the rest of the spec though!

Reply:
Lgtm! I'd say ship it!